### PR TITLE
*: run chaos-mesh operations with GitHub action

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

- run chaos-mesh operations with GitHub action

### What is changed and how it works?

- add a chaos-mesh GitHub action workflow
- add a test case with chaos-mesh operations

NOTE：
- no DDL added in the sample case and should be added later
- only one source in the sample case and should add more sources and shard merge cases

### Check List 

Tests 

 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: .github/workflows/chaos-mesh.yml [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+    - name: Run chaos mesh action
+      uses: chaos-mesh/chaos-mesh-action@master
+      env:
+        CFG_BASE64: ${{ env.CFG_BASE64 }}
+
+    # TODO: use timeout to start the binary of chaos-case.
+    - name: Wait for chaos test case complete
+      run: |
+        kubectl wait --for=condition=complete job/chaos-test-case --timeout=140s
```
Comment (csuzhangxc) [inline .github/workflows/chaos-mesh.yml:217]: I'll update this `--timeout` to a long time before PR merging.

File: chaos/manifests/cases.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+          ports:
+            - containerPort: 80
+              name: chaos-test-case
+          command:
+            - "/chaos-case"
+            - "--config-dir=/" # we put config files into the root directory when building the docker image
+            - "--duration=120s" # TODO: 120s only for dev
```
Comment (csuzhangxc) [inline chaos/manifests/cases.yaml:27]: I'll update this `--duration` to a long time before PR merging.

File: .github/workflows/chaos-mesh.yml [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        kubectl get -f $GITHUB_WORKSPACE/chaos/manifests/cases.yaml
+        kubectl describe -f $GITHUB_WORKSPACE/chaos/manifests/cases.yaml
+        sleep 60
+
+    - name: Encode chaos-mesh action
+      run: |
+        echo "::set-env name=CFG_BASE64::$(base64 -w 0 $GITHUB_WORKSPACE/chaos/manifests/pod-failure-dm.yaml)"
+
+    - name: Run chaos mesh action
+      uses: chaos-mesh/chaos-mesh-action@master
```
Comment (lance6716) [inline .github/workflows/chaos-mesh.yml:210]: Not sure where GitHub action calls `chaos/cases`

**Comment by lance6716** [comment]
LGTM

File: .github/workflows/chaos-mesh.yml (hunk 3 above)
Comment (csuzhangxc) [reply .github/workflows/chaos-mesh.yml:210]: `uses: chaos-mesh/chaos-mesh-action@master` only applis chaos operations (defined in `./chaos/manifests/pod-failure-dm.yaml`) into the cluster.

`chaos/cases` is called by `./chaos/manifests/cases.yaml` (see https://github.com/pingcap/dm/blob/87f60a3cf33f1fe588d9b420e8169124934d66f9/chaos/manifests/cases.yaml#L25-L27) ( and `cases.yaml`  is applied in `kubectl apply -f $GITHUB_WORKSPACE/chaos/manifests/cases.yaml`).

**Review by GMHDBJD — DISMISSED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]

**Comment by csuzhangxc** [comment]
@lance6716 @GMHDBJD 

I did some updates after your review:
- send Slack message if the test case failed (only to me now)
- replace KIND with K3s as the K8s cluster for performance and usability
- add tzdata into the docker image
- fix selector usage for chaos-mesh (it has different selector with K8s)
- update logs as artifact before ending the workflow (see a sample in https://github.com/pingcap/dm/actions/runs/266604455)

PTAL again

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**